### PR TITLE
REFACTOR: rewrite cci indicator into v2

### PR DESCRIPTION
**Pull request by c9s**

### CODE REVIEW

File: pkg/types/kline.go [hunk 1]
```diff
@@ -10,14 +10,15 @@ import (
 	"github.com/c9s/bbgo/pkg/style"
 )
 
+var Two = fixedpoint.NewFromInt(2)
+var three = fixedpoint.NewFromInt(3)
```
Comment (zenixls2) [inline pkg/types/kline.go:14]: Three

File: pkg/types/series_float64.go [hunk 2]
```diff
@@ -52,13 +58,17 @@ func (f *Float64Series) Bind(source Float64Source, target Float64Calculator) {
 	if canTruncate {
 		c = func(x float64) {
 			y := target.Calculate(x)
-			target.PushAndEmit(y)
-			trc.Truncate()
+			if !math.IsNaN(y) {
```
Comment (zenixls2) [inline pkg/types/series_float64.go:61]: I don't think Float64Series should do such filtering.
A Filter function or Map function will do the work.

File: pkg/types/series_float64.go [hunk 3]
```diff
@@ -52,13 +58,17 @@ func (f *Float64Series) Bind(source Float64Source, target Float64Calculator) {
...
+				target.PushAndEmit(y)
+				trc.Truncate()
+			}
 		}
 	} else {
 		c = func(x float64) {
 			y := target.Calculate(x)
-			target.PushAndEmit(y)
+			if !math.IsNaN(y) {
```
Comment (zenixls2) [inline pkg/types/series_float64.go:69]: same here

File: pkg/types/series_float64.go (hunk 2 above)
Comment (c9s) [reply pkg/types/series_float64.go:61]: Ok~

File: pkg/types/kline.go (hunk 1 above)
Comment (c9s) [reply pkg/types/kline.go]: Updated

File: pkg/types/series_float64.go (hunk 2 above)
Comment (c9s) [reply pkg/types/series_float64.go]: Updated

File: pkg/types/series_float64.go (hunk 3 above)
Comment (c9s) [reply pkg/types/series_float64.go]: Updated

**Review by zenixls2 — APPROVED** [review]

**Review by andycheng123 — APPROVED** [review]

**Pull request merged**